### ISSUE TITLE
Responsive UI and styles

### ISSUE THREAD

**Issue by llkats** [nodeschool/sanfrancisco#182]
Decide upon a CSS framework (if desired)!

**Comment by teone** [comment]
I agree with being light weight (as discussed in the Website refresh issue) but since we already need a build system I lean towards using a preprocessor

**Comment by skipzero** [comment]
You had mentioned using gulp in another issue. Are there css preprocessors for gulp?

**Comment by teone** [comment]
Of course :) I'm sure about scss, sass, stylus and less... I don't have a strong preference here, just more familiar with sass/scss

**Comment by teone** [comment]
If we want to keep things minimal we can just use http://neat.bourbon.io/ to build a responsive grid (it basically provide a collection of `sass mixins`  that you can use to create your own grid

**Comment by tlhunter** [comment]
I vote against using a CSS preprocessor. The people who know SCSS is a subset of the number of people who know CSS and getting contributors to help with the site should be our highest priority.

**Comment by teone** [comment]
I would argue that any valid `css` is also a valid `scss`, so it shouldn't be an issue and that can be a good opportunity to learn about CSS preprocessors, but if you feel that it would limit our possibilities to find contributors I'm definitely ok with plain css.

**Comment by llkats** [comment]
Hi friends, I've largely been unopinionated on these decisions mostly because I lack the bandwidth to form a strong opinion. Thus I'm biased towards the desires of those who want to submit PRs :) Does a decision need to be made in order to move forwards?

**Comment by skipzero** [comment]
So why don't we just use bootstrap from a CDN? that'll take care of hassle/deciding between css/sass... Then we can get busy getting busy and not talk about getting busy. ;D. 
Since it's pretty much needed before the other sections are built out, I vote for easy, quick and working!
:D

**Comment by teone** [comment]
I think the only point agains bootstrap was the `lightweight` consideration, but I'm ok with it.

**Comment by llkats** [comment]
We had some discussion around this on Saturday. I think that we were just brainstorming ideas, but hadn't come to any real conclusions about what to actually use.

The "lightweight" consideration doesn't just refer to the project's size, or to the page load size. I also mean the amount of prior knowledge one needs to get running with a project. If one needs to learn an additional API or three, then that's a roadblock to contributing.

The benefits of leaning on a framework like Bourbon or Bootstrap are, to me, having some naming conventions and initial styles already in place. Things like grid systems are difficult to bolt on after the fact, so it's nice to define these from the get-go. And since this project is rather grassroots in nature, without a ton of design direction, having some styles in place could help us initially.

The negatives are the additional asset size, as well as potential collaborators being unfamiliar with that project's conventions and having to digest that before getting up to speed. It also might be bothersome to convert a project from one scheme to another, or to override styles with our custom design, or to rip out altogether later on.

Does anyone else have any thoughts on these things before a UI framework is chosen? I just want to be deliberate in how we choose things, come to a consensus, while striking a balance against unnecessary bikeshedding.

**Comment by skipzero** [comment]
I'm cool with Bourbon over BS, it's lighter in both respects, I'd imagine. I've never used it, so it'd be a little fun to have a reason to. if people aren't comfortable with sass, it should still be fine to do long hand in css. and I haven't used bourbon/neat, so that could be a fun experience, personally speaking.

**Comment by teone** [comment]
@skipzero that was my point in favor of using Bourbon, I've always looked at it but never had a real opportunity to use it :)

@llkats I agree that someone that is not familiar with one of the should climb one more step to be able to contribute, but if we pick up something that is spread as bootstrap is, they'll have to climb this step anyway soon or later if they want to be in the web business :) so why not now?

That said, I would propose to use https://github.com/twbs/bootstrap-sass, it will give us:
- lightweight in term of file size (we can import only the grid if that's all wee need)
- lightweight in term of needed knowledge 

How do you feel about it?

**Comment by skipzero** [comment]
I'm gonna go ahead and run with BS, then. I can integrate that into the work you've done already @teone or you're welcome to handle that in which case I'd basically just be linking to the built files...

**Comment by teone** [comment]
@skipzero let me know if I can help, but integrating `sass` with `gulp + browserSync` is pretty straight forward :) and well documented on BrowserSync website's: https://browsersync.io/docs/gulp#gulp-sass-css

**Comment by skipzero** [comment]
Yeah, I literally just did it at work last week! ;)
that's what I find funny about all the build tool/framework talk lately. it's mirrored my job.

**Comment by llkats** [comment]
Sounds great, carry on! Thanks!

**Comment by skipzero** [comment]
Sass and TWBS added. I was having issues with a rebase at the beginning so hopefully I didn't mess anything up too badly. everything seems to be working and we've got our styles as well as bootstraps. 

In the interest of simplicity and whatnot, I just downloaded TWBS, unzipped and added into a vendors folder in source to avoid some of the hassle that is Bower and that whole other can o' worms.

If anyone catches anything, I'm happy to fix. Let me know if you have any questions about anything, as well.  

We should now be pretty close to building out the site! Sorry for taking so long, my office is moving and the last couple weeks were that on top of the usual amount of crazy. Should be a little less nuts now... I hope!

**Comment by skipzero** [comment]
I've pushed another change and I believe took care of everything found that needed to be. Take a look, please. I'm leaving the history this time and will make a note to squash history before the next PR. If i've missed anything, please let me know!
